src/mbgl/renderer/painter.hpp

```diff
@@ -178,7 +177,7 @@ class Painter : private util::noncopyable {
     }();
 
 private:
-    TransformState state;
+    Transform& transform;
```

Comment:
Yeah, that's my one worry about this PR, the way we "poke through" the transform to grab the state. The painter needs to know the conversion (as put in originally in your background stuff in 4603ca0), but perhaps this can be pushed down to the state, then have the public map APIs poke through map to transform to transform state to get them. Will look at this.